docker/Dockerfile

```diff
@@ -0,0 +1,49 @@
+# build from the root of this repo:
+# docker build -t gcr.io/cloud-devrel-kokoro-resources/synthtool -f docker/Dockerfile .
+FROM python:3.6.12-buster
+
+WORKDIR /
+
+######################## Install nodejs.
+# Use `nvm` to bootstrap the installation of node.js and npm.
+# To learn more: https://github.com/nvm-sh/nvm
+RUN curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.35.3/install.sh | bash
+
+# Use the latest LTS version of nodejs.  This can change over time, but
+# should prevent the need to modify this file every year.
+# The "npm config set unsafe-perm true" is necessary because npm will refuse to perform
+# some operations if the user is running as root.  The root user of a docker image
+# doesn't have root permissions on the machine, so it's safe to execute them.
+# I learned this from https://geedew.com/What-does-unsafe-perm-in-npm-actually-do/
+RUN /bin/bash -c "source /root/.nvm/nvm.sh && nvm install --lts && npm config set unsafe-perm true"
+
+####################### Install bazel 3.0.0
+RUN mkdir -p /bazel
+RUN curl -L https://github.com/bazelbuild/bazel/releases/download/3.0.0/bazel-3.0.0-linux-x86_64 -o /bazel/bazel
+RUN chmod +x /bazel/bazel
```

Comment:
If we're worried about optimizing the image size (number of layers), we should combine `RUN` statements with `&&`.

Reply:
I'm not yet worried.  What symptoms will I observe when there are too many layers?

Reply:
Bigger image sizes and longer downloads (each layer is pulled and decompressed).